### ISSUE TITLE
C/C++ extension make the os extremely slow, not cpu usage with IntelliSense caching

### ISSUE THREAD

**Issue by Ziaeemehr** [microsoft/vscode-cpptools#3433]
Type: General

**Describe the bug**
- OS and Version: Ubuntu 18.04, Mint 18.3
- VS Code Version: 1.32
- C/C++ Extension Version: 0.22.1

when activate the C/C++ extension after a while the OS become extremely slow, and again back to normal state after few minuets. htop does not show any CPU usage and there is enough free RAM. I double checked to make sure this is because of the extension.

**Comment by sean-mcmanus** [comment]
Are the Microsoft.VSCode.CPP.* processes using a lot of memory?

Does setting the C_Cpp.intelliSenseCache size to 0 fix the issue?

Does the slowness occur when the red flame appears in the bottom right?

**Comment by Ziaeemehr** [comment]
Are the Microsoft.VSCode.CPP.* processes using a lot of memory? NO
Does setting the C_Cpp.intelliSenseCache size to 0 fix the issue? YES
Does the slowness occur when the red flame appears in the bottom right? NO

**Comment by sean-mcmanus** [comment]
Okay -- yeah, setting the intelliSenseCache size is the recommended workaround until we can figure out why some users are hitting this problem.

The IntelliSense cache can write a large amount of data to your disk drive -- it's intended to improve performance.

Is there something special about your file system or disk drive that would make using large files slow?

Does the issue occur for certain repro projects?

We're trying to figure out why we're not seeing OS slowdowns (we've gotten a handful of reports like this).

On Windows there is ETW system tracing that generates ETL logs that can be used to diagnose system performance problems, but we still need to investigate an equivalent on Linux (ptrace?).

**Comment by Ziaeemehr** [comment]
`Is there something special about your file system or disk drive that would make using large files slow?`
No I don't think so.  
`Does the issue occur for certain repro projects?`
I tried it for several projects, It happened for all of them.
Thank you for your answer and consideration.

**Comment by kewarken** [comment]
FYI I'm seeing something similar running Ubuntu 18.04 in a VM. I had to completely uninstall the extension to make the editor usable. Attempting to set the cache size to zero to see if it helps.

**Comment by caodingpeng** [comment]
i also met this problem. vscode is freezes when c/c++ plugin activated

**Comment by F4FXL** [comment]
Same here on debian sid running code-insiders. Setting cache to 0 does not help at all.

**Comment by sean-mcmanus** [comment]
@F4FXL You're seeing our extension not using CPU, but the OS becomes slow? I don't think we've ever gotten a repro of this.

**Comment by F4FXL** [comment]
> @F4FXL You're seeing our extension not using CPU, but the OS becomes slow? I don't think we've ever gotten a repro of this.

I investigated further, it was actually the tracker file indexation going crazy when VS Code is running. I turned of file monitor and excluded the folders containings my projects and all is fine now.
![image](https://user-images.githubusercontent.com/5264350/79060923-ce7f4600-7c8a-11ea-9235-c4232eb11d22.png)

However, generally speaking it seems VS Code is making a lot of file accesses in the background, but this in another topic.
